x-pack/plugins/aiops/server/routes/explain_log_rate_spikes.ts

```diff
@@ -253,202 +278,232 @@ export const defineExplainLogRateSpikesRoute = (
+                        term: { [d.fieldName]: d.fieldValue },
+                      })),
+                    },
                   };
-                }) ?? [];
 
-              push(
-                addChangePointsGroupHistogramAction([
-                  {
-                    id: cpg.id,
-                    histogram,
-                  },
-                ])
-              );
-            });
+                  let cpgTimeSeries: NumericChartData;
+                  try {
+                    cpgTimeSeries = (
+                      (await fetchHistogramsForFields(
+                        client,
+                        request.body.index,
+                        histogramQuery,
+                        // fields
+                        [
+                          {
+                            fieldName: request.body.timeFieldName,
+                            type: KBN_FIELD_TYPES.DATE,
+                            interval: overallTimeSeries.interval,
+                            min: overallTimeSeries.stats[0],
+                            max: overallTimeSeries.stats[1],
+                          },
+                        ],
+                        // samplerShardSize
+                        -1,
+                        undefined
+                      )) as [NumericChartData]
+                    )[0];
+                  } catch (e) {
+                    logger.error(
+                      `Failed to fetch the histogram data for group #${
+                        cpg.id
+                      }, got: \n${e.toString()}`
+                    );
+                    pushError(`Failed to fetch the histogram data for group #${cpg.id}.`);
+                    return;
+                  }
+                  const histogram =
+                    overallTimeSeries.data.map((o, i) => {
+                      const current = cpgTimeSeries.data.find(
+                        (d1) => d1.key_as_string === o.key_as_string
+                      ) ?? {
+                        doc_count: 0,
+                      };
+                      return {
+                        key: o.key,
+                        key_as_string: o.key_as_string ?? '',
+                        doc_count_change_point: current.doc_count,
+                        doc_count_overall: Math.max(0, o.doc_count - current.doc_count),
+                      };
+                    }) ?? [];
+
+                  push(
+                    addChangePointsGroupHistogramAction([
+                      {
+                        id: cpg.id,
+                        histogram,
+                      },
+                    ])
+                  );
+                }
+              });
+            }
+          } catch (e) {
+            logger.error(
+              `Failed to transform field/value pairs into groups, got: \n${e.toString()}`
+            );
+            pushError(`Failed to transform field/value pairs  into groups.`);
```

Comment:
typo: there's an extra space here after 'pairs'

Reply:
Fixed in [c9eed74](https://github.com/elastic/kibana/pull/142047/commits/c9eed74873e561135c156bc028cc056f84e19965).